test/e2e_node/resource_collector.go

```diff
@@ -371,18 +372,21 @@ func getCadvisorPod() *v1.Pod {
 // deletePodsSync deletes a list of pods and block until pods disappear.
 func deletePodsSync(f *framework.Framework, pods []*v1.Pod) {
 	var wg sync.WaitGroup
-	for _, pod := range pods {
+	for i := range pods {
+		pod := pods[i]
 		wg.Add(1)
-		go func(pod *v1.Pod) {
+		go func() {
 			defer ginkgo.GinkgoRecover()
 			defer wg.Done()
 
 			err := f.PodClient().Delete(context.TODO(), pod.ObjectMeta.Name, *metav1.NewDeleteOptions(30))
-			framework.ExpectNoError(err)
+			if apierrors.IsNotFound(err) {
+				framework.Failf("Unexpected error trying to delete pod %s: %v", pod.Name, err)
+			}
 
 			gomega.Expect(e2epod.WaitForPodToDisappear(f.ClientSet, f.Namespace.Name, pod.ObjectMeta.Name, labels.Everything(),
 				30*time.Second, 10*time.Minute)).NotTo(gomega.HaveOccurred())
-		}(pod)
+		}()
```

Comment:
hm, thinking about it, this is not a problem. When pod is passed as argument here, things should be fine

Reply:
we can keep either new format or revert, shouldn't make a difference